pkg/dyninst/decode/decoder.go

```diff
@@ -107,21 +124,9 @@ type typeAndAddr struct {
 }
 
 func (d *Decoder) Decode(event output.Event, out io.Writer) error {
```

Comment:
I've changed my mind on what I want here. Instead of having the ProcessInfoResolver, let's instead define a structure and pass this as the argument to `Decode`

```go
package decode

type Event struct {
    output.Event
    ServiceName string
    // Whatever else we need...
}
```